lib/sycamore/sycamore/utils/nested.py

```diff
@@ -0,0 +1,18 @@
+from typing import Any
+
+
+def nested_lookup(d: Any, keys: list[str]) -> Any:
+    # Eventually we can support integer indexes into tuples and lists also
+    while len(keys) > 0:
+        if not isinstance(d, dict):
+            return None
+        if keys[0] not in d:
+            return None
+        d = d[keys[0]]
+        keys = keys[1:]
+
+    return d
+
+
+def dotted_lookup(d: Any, keys: str) -> Any:
```

Comment:
can you consolidate with https://github.com/aryn-ai/sycamore/blob/23747b1854523d38c785950f4ac9ec78307eec4b/lib/sycamore/sycamore/data/document.py#L217 ? 2 methods doing similar stuff

Reply:
Done. I thought I had a todo to do that. Turned out to be not too hard.